clippy_lints/src/methods/ptr_offset_with_cast.rs

```diff
@@ -0,0 +1,77 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::source::SpanRangeExt;
+use clippy_utils::sym;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_span::Symbol;
+use std::fmt;
+
+use super::PTR_OFFSET_WITH_CAST;
+
+pub(super) fn check(cx: &LateContext<'_>, method: Symbol, expr: &Expr<'_>, recv: &Expr<'_>, arg: &Expr<'_>) {
+    let ty::RawPtr(_, _) = cx.typeck_results().expr_ty(recv).kind() else {
+        return;
+    };
```

Comment:
You can use `Ty::is_raw_ptr()` here.

```suggestion
    if !cx.typeck_results().expr_ty(recv).is_raw_ptr() {
        return;
    };
```

Also, is there a reason not to use `expr_ty_adjusted()` here? For example, the lint should trigger if `.offset(… as _)` if used on a `&*const T` as well as a `*const T`.